From d8500b3b451aca9f55712579d63293b9c000ffe5 Mon Sep 17 00:00:00 2001
From: Anibal Portero Hermida <anibal.portero@pantacor.com>
Date: Tue, 3 Mar 2026 09:56:09 +0000
Subject: [PATCH] feat(lxc6-pv): add lxc6 patch to fix infinite loop on closeall

---
 src/lxc/start.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/src/lxc/start.c b/src/lxc/start.c
index e0ec0b6f4..c4ad10e56 100644
--- a/src/lxc/start.c
+++ b/src/lxc/start.c
@@ -238,7 +238,7 @@ static unsigned int get_listen_fds_max(void)
 int lxc_check_inherited(struct lxc_conf *conf, bool closeall,
 			int *fds_to_ignore, size_t len_fds)
 {
-	int fd, fddir;
+	int fd, fddir, prev_fd = -1;
 	size_t i;
 	DIR *dir;
 	struct dirent *direntp;
@@ -284,6 +284,11 @@ restart:
 			if (fds_to_ignore[i] == fd)
 				break;
 
+		if (prev_fd == fd) {
+			WARN("Tried already to close inherited fd %d. Ignoring...", fd);
+			break;
+		}
+
 		if (fd == fddir || fd == lxc_log_fd || fd == lxc_log_out_fd ||
 		    (i < len_fds && fd == fds_to_ignore[i]))
 			continue;
@@ -320,6 +325,7 @@ restart:
 		}
 
 		if (closeall) {
+			prev_fd = fd;
 			if (close(fd))
 				SYSINFO("Closed inherited fd %d", fd);
 			else
