diff --git a/meta-bsp/recipes-bsp/imx-mkimage/imx-boot_1.0.bb b/meta-bsp/recipes-bsp/imx-mkimage/imx-boot_1.0.bb
index 451cce8365..b78fc47eed 100644
--- a/meta-bsp/recipes-bsp/imx-mkimage/imx-boot_1.0.bb
+++ b/meta-bsp/recipes-bsp/imx-mkimage/imx-boot_1.0.bb
@@ -28,6 +28,7 @@ inherit deploy uuu_bootloader_tag
 
 UUU_BOOTLOADER        = "${BOOT_NAME}"
 UUU_BOOTLOADER_TAGGED = "${BOOT_NAME}-tagged"
+UUU_BOOTLOADER_UNTAGGED = "${BOOT_NAME}-untagged"
 
 # Add CFLAGS with native INCDIR & LIBDIR for imx-mkimage build
 CFLAGS = "-O2 -Wall -std=c99 -I ${STAGING_INCDIR_NATIVE} -L ${STAGING_LIBDIR_NATIVE}"
