From 6c9554dd1e6cbae390f8326448d98bf4288d96e9 Mon Sep 17 00:00:00 2001
From: Alexander Sack <asac@pantacor.com>
Date: Wed, 11 May 2022 17:00:13 +0200
Subject: [PATCH] add support for FOLLOW_X_ROOT env; superseed ROOTFSDIR in
 turn

Upstream-Status: Pending

---
 util-linux/mdev.c | 73 +++++++++++++++++++++++++++++++++++++----------
 1 file changed, 58 insertions(+), 15 deletions(-)

diff --git a/util-linux/mdev.c b/util-linux/mdev.c
index f41945d..3c96030 100644
--- a/util-linux/mdev.c
+++ b/util-linux/mdev.c
@@ -74,6 +74,17 @@
 //config:	hotplug events this mode of operation will consume less
 //config:	resources than registering mdev as hotplug helper or using the
 //config:	uevent applet.
+//config:
+//config:config FEATURE_MDEV_FOLLOW_X_ROOT
+//config:	bool "Support following a switch or pivot root in mount_namespace"
+//config:	default y
+//config:	depends on MDEV
+//config:	help
+//config:	Add support for following a switch_root and pivot_root of pid 1
+//config:	Useful for having mdev prefer the dev/ nodes of a root getting
+//config:	prepared in a mount namespace before the pivot/switch_root and then
+//config:	continue operating in DAEMON mode inside mnt namespace afterwards.
+//config:
 
 //applet:IF_MDEV(APPLET(mdev, BB_DIR_SBIN, BB_SUID_DROP))
 
@@ -116,6 +127,7 @@
 #include "xregex.h"
 #include <linux/netlink.h>
 #include <syslog.h>
+#include <unistd.h>
 
 /* "mdev -s" scans /sys/class/xxx, looking for directories which have dev
  * file (it is of the form "M:m\n"). Example: /sys/class/tty/tty0/dev
@@ -322,6 +334,11 @@ struct globals {
 /* We use additional bytes in make_device() */
 #define SCRATCH_SIZE 128
 
+#if ENABLE_FEATURE_MDEV_FOLLOW_X_ROOT
+static char *newroot = NULL;
+static char *newrootdev = NULL;
+#endif
+
 #if ENABLE_FEATURE_MDEV_CONF
 
 static void make_default_cur_rule(void)
@@ -575,6 +592,25 @@ static void make_device(char *device_name, char *path, int operation)
 {
 	int major, minor, type, len;
 	char *path_end = path + strlen(path);
+	struct stat s;
+	static char wd [PATH_MAX] = {0};
+
+#if ENABLE_FEATURE_MDEV_FOLLOW_X_ROOT
+	if (newrootdev && !(!access(newrootdev, X_OK) && !stat(newrootdev, &s) && (S_IFDIR & s.st_mode))){
+		dbg1("following x root...");
+		newrootdev=NULL;
+		chdir("/");
+		chdir("..");
+		chroot(".");
+	}
+
+	if (newrootdev)
+		chdir(newrootdev);
+	else
+#endif
+		chdir ("/dev");
+
+	getcwd(wd, PATH_MAX);
 
 	/* Try to read major/minor string.  Note that the kernel puts \n after
 	 * the data, so we don't need to worry about null terminating the string
@@ -893,18 +929,6 @@ static int FAST_FUNC dirAction(struct recursive_state *state,
 	return (state->depth >= MAX_SYSFS_DEPTH ? SKIP : TRUE);
 }
 
-static const char *getdevdir() {
-	static char *devdir = NULL;
-
-	if (!devdir) {
-		const char *rootfsdir = getenv("ROOTFSDIR");
-		devdir = malloc (sizeof(char) * PATH_MAX);
-		sprintf(devdir, "%s%s", rootfsdir ? rootfsdir : "", "/dev");
-	}
-	return devdir;
-}
-
-
 /* For the full gory details, see linux/Documentation/firmware_class/README
  *
  * Firmware loading works like this:
@@ -965,7 +989,13 @@ static void load_firmware(const char *firmware, const char *sysfs_path)
 		full_write(loading_fd, "-1", 2);
 
  out:
-	xchdir(getdevdir());
+#if ENABLE_FEATURE_MDEV_FOLLOW_X_ROOT
+	if (newrootdev)
+		xchdir(newrootdev);
+	else
+#endif
+		xchdir("/dev");
+
 	if (ENABLE_FEATURE_CLEAN_UP) {
 		close(firmware_fd);
 		close(loading_fd);
@@ -1268,14 +1298,27 @@ int mdev_main(int argc UNUSED_PARAM, char **argv)
 	/* Force the configuration file settings exactly */
 	umask(0);
 
-	xchdir(getdevdir());
-
 	opt = getopt32(argv, "^"
 		"sS" IF_FEATURE_MDEV_DAEMON("df") "v"
 		"\0"
 		"vv",
 		&G.verbose);
 
+#if ENABLE_FEATURE_MDEV_FOLLOW_X_ROOT
+	newroot = getenv("FOLLOW_X_ROOT");
+
+	if (newroot && !newrootdev) {
+		newrootdev = malloc(sizeof(char) * (strlen(newroot) + 6));
+		sprintf(newrootdev, "%s%s", newroot, "/dev");
+	}
+
+	if (newrootdev)
+		xchdir(newrootdev);
+	else
+#endif
+		xchdir("/dev");
+
+
 #if ENABLE_FEATURE_MDEV_CONF
 	G.filename = getenv("MDEV_CONF") ? getenv("MDEV_CONF") : "/etc/mdev.conf";
 	if (opt & (MDEV_OPT_SCAN|MDEV_OPT_DAEMON)) {
-- 
2.43.0

