diff --git a/bufferevent.c b/bufferevent.c
index 79f76f4e..6bd31a94 100644
--- a/bufferevent.c
+++ b/bufferevent.c
@@ -748,6 +748,14 @@ bufferevent_decref_and_unlock_(struct bufferevent *bufev)
 	return 1;
 }
 
+void
+bufferevent_set_finalize_cb(struct bufferevent *bufev,
+    void (*cb)(struct bufferevent *, void *), void *arg)
+{
+	bufev->finalize_cb = cb;
+	bufev->finalize_arg = arg;
+}
+
 static void
 bufferevent_finalize_cb_(struct event_callback *evcb, void *arg_)
 {
@@ -798,6 +806,11 @@ bufferevent_finalize_cb_(struct event_callback *evcb, void *arg_)
 	 */
 	if (underlying)
 		bufferevent_decref_(underlying);
+
+	if ((*bufev->finalize_cb)) {
+		(*bufev->finalize_cb)(bufev, bufev->finalize_arg);
+		bufev->finalize_cb = NULL;
+	}
 }
 
 int
diff --git a/include/event2/bufferevent.h b/include/event2/bufferevent.h
index 68c3f264..43f13ce5 100644
--- a/include/event2/bufferevent.h
+++ b/include/event2/bufferevent.h
@@ -326,6 +326,17 @@ int bufferevent_get_priority(const struct bufferevent *bufev);
 EVENT2_EXPORT_SYMBOL
 void bufferevent_free(struct bufferevent *bufev);
 
+/**
+  Changes the finalize callback for a bufferevent.
+
+  @param bufev the bufferevent object for which to change the callback. bufev must not be NULL
+  @param cb callback to invoke when bufferevent is finialized with no pending events
+  @param arg an argument that will be supplied to the callback
+  @see bufferevent_new()
+**/
+EVENT2_EXPORT_SYMBOL
+void bufferevent_set_finalize_cb(struct bufferevent *bufev,
+    void (*cb)(struct bufferevent *, void *), void *arg);
 
 /**
   Changes the callbacks for a bufferevent.
diff --git a/include/event2/bufferevent_struct.h b/include/event2/bufferevent_struct.h
index 546773ab..3ce9f683 100644
--- a/include/event2/bufferevent_struct.h
+++ b/include/event2/bufferevent_struct.h
@@ -107,6 +107,10 @@ struct bufferevent {
 	/** Events that are currently enabled: currently EV_READ and EV_WRITE
 	    are supported. */
 	short enabled;
+
+	/** Callback to be called after bufferevent has finalized with no further events */
+	void (*finalize_cb)(struct evhttp_request *, void *);
+	void *finalize_arg;
 };
 
 #ifdef __cplusplus
